### ISSUE TITLE
Make variable secured only when it's in 'Comma separated list of outputs to secure'

### ISSUE THREAD

**Issue by yar-shukan** [rwatjen/ARMOutputParserVSTSExtension#2]
Having all variables being output as secured has some important drawback: log readability and issues trace ability. Example: 'App Service '***-staging' started successfully.'

It becomes a nightmare to understand what's deployed/updated by following such logs. **Especially when there's nothing secured in outputs and these outputs are used across the log.**

Proposal:

1. Make ;issecret=true/false configurable with the following behavior:

- by default ;issecret has to be false
- add the "black list" variable like "Comma separated list of outputs to secure". if the variable is in this list - it will be set as  ;issecret=true

2. As it's a breaking change in design/behavior release this as version 2.0.0 not to break existing users and put the change in release notes.

From my experience with ARM output most of the time output variables doesn't have any sensitive information so it doesn't make sense to secure them out. If there're variables that are sensitive - that has to be outlined explicitly by "Comma separated list of outputs to secure".

@rwatjen Please let me know what you think. If you give green-light for this - I can handle the change. Otherwise we'll have to switch to maybe our own implementation as with '***' for everything in log it becomes unreadable for people and it's a challenge to troubleshoot releases.

**Comment by rwatjen** [comment]
I agree that it is annoying that all outputs are logged as *** instead of their value, when they are not secrets.

Adding a comma separated list to manage is a good idea. It would be even better if the VSTS integration would support ARM `securestring` and `secureObject`, but it doesn't. That's why I decided to mark all output as secret.

As you mentioned, changing it would be a behavioral breaking change. I prefer erring on the side of caution, so if it should be implemented, I think the comma separated list should contain those variables which should **not** be marked as secret.

If you want, it would be awesome, if you created a PR to do the following:

*  Make ;issecret=true/false configurable with the following behavior:
by default ;issecret has to be **true**
* Add the "exclude list" variable like "Comma separated list of outputs to mark as **non**-secret". if the variable is in this list - it will be set as ;issecret=**false**

That way, it also would not be a behavior breaking change, and would not require a 2.x, but could instead be a 1.1 update.

**Comment by billchesnut** [comment]
I really need some way to not make all of the values secret, causing lots of problems for what I am trying to do with it.

**Comment by gpadavala** [comment]
need this, can't even find a way to print the damn thing :(

**Comment by sandromastronardi** [comment]
It's bad practice to output secrets in arm templates anyways since they are visible in the Azure Portal.
A boolean (that is default to true) that can be set to false will not break the behaviour of previous version, but it makes this plugin 100 times more usable.. ;-) A list is not really needed imho as it is a malpractice to output secrets (store them to the vault)..
